src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1896,20 +1896,25 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_RequestSEDActiv
 
     if (!onOff && delayIdle && CHIP_DEVICE_CONFIG_SED_ACTIVE_THRESHOLD.count() != 0)
     {
-        err = DeviceLayer::SystemLayer().StartTimer(CHIP_DEVICE_CONFIG_SED_ACTIVE_THRESHOLD, RequestSEDModeChange, this);
+        err = DeviceLayer::SystemLayer().StartTimer(CHIP_DEVICE_CONFIG_SED_ACTIVE_THRESHOLD, RequestSEDModeUpdate, this);
         if (CHIP_NO_ERROR == err)
         {
+            if (!mDelayIdleTimerRunning)
+            {
+                mDelayIdleTimerRunning = true;
+                mActiveModeConsumers++;
+            }
             return err;
         }
 
-        ChipLogError(DeviceLayer, "Failed to postponed Idle Mode with error %" CHIP_ERROR_FORMAT, err.Format());
+        ChipLogError(DeviceLayer, "Failed to postpone Idle Mode with error %" CHIP_ERROR_FORMAT, err.Format());
     }
 
-    return SEDChangeMode();
+    return SEDUpdateMode();
```

Comment:
What if StartTimer fails but mDelayIdleTimerRunning is true? This means SEDUpdateMode will be called twice right?

I am not sure this can ever happen ... but just wondering. Maybe a comment of 'we do not care/never happens'

Reply:
Update will be called twice indeed but since the timer counts as an active consumer, the first update won't trigger a Mode change.